src/Evolution/Systems/RadiationTransport/M1Grey/M1HydroCoupling.hpp

```diff
@@ -0,0 +1,165 @@
+      Tags::M1HydroCouplingNormal<NeutrinoSpecies>...,
+      Tags::M1HydroCouplingSpatial<Frame::Inertial, NeutrinoSpecies>...>;
+
+  using argument_tags =
+      tmpl::list<Tags::GreyEmissivity<NeutrinoSpecies>...,
+                 Tags::GreyAbsorptionOpacity<NeutrinoSpecies>...,
+                 Tags::GreyScatteringOpacity<NeutrinoSpecies>...,
+                 Tags::TildeJ<NeutrinoSpecies>...,
+                 Tags::TildeHNormal<NeutrinoSpecies>...,
+                 Tags::TildeHSpatial<Frame::Inertial, NeutrinoSpecies>...,
+                 hydro::Tags::SpatialVelocity<DataVector, 3, Frame::Inertial>,
+                 hydro::Tags::LorentzFactor<DataVector>, gr::Tags::Lapse<>,
+                 gr::Tags::SpatialMetric<3>, gr::Tags::SqrtDetSpatialMetric<>>;
+
+  static void apply(
+      const gsl::not_null<db::item_type<
+          Tags::M1HydroCouplingNormal<NeutrinoSpecies>>*>... source_n,
+      const gsl::not_null<db::item_type<Tags::M1HydroCouplingSpatial<
+          Frame::Inertial, NeutrinoSpecies>>*>... source_i,
+      const db::const_item_type<
+          Tags::GreyEmissivity<NeutrinoSpecies>>&... emissivity,
+      const db::const_item_type<
+          Tags::GreyAbsorptionOpacity<NeutrinoSpecies>>&... absorption_opacity,
+      const db::const_item_type<
+          Tags::GreyScatteringOpacity<NeutrinoSpecies>>&... scattering_opacity,
+      const db::const_item_type<Tags::TildeJ<NeutrinoSpecies>>&... tilde_j,
+      const db::const_item_type<
+          Tags::TildeHNormal<NeutrinoSpecies>>&... tilde_hn,
+      const db::const_item_type<
+          Tags::TildeHSpatial<Frame::Inertial, NeutrinoSpecies>>&... tilde_hi,
+      const db::const_item_type<
+          hydro::Tags::SpatialVelocity<DataVector, 3, Frame::Inertial>>&
+          spatial_velocity,
+      const db::const_item_type<hydro::Tags::LorentzFactor<DataVector>>&
+          lorentz_factor,
+      const db::const_item_type<gr::Tags::Lapse<>>& lapse,
+      const tnsr::ii<DataVector, 3, Frame::Inertial>& spatial_metric,
+      const db::const_item_type<gr::Tags::SqrtDetSpatialMetric<>>&
+          sqrt_det_spatial_metric) noexcept {
+    EXPAND_PACK_LEFT_TO_RIGHT(detail::compute_m1_hydro_coupling_impl(
+        source_n, source_i, emissivity, absorption_opacity, scattering_opacity,
+        tilde_j, tilde_hn, tilde_hi, spatial_velocity, lorentz_factor, lapse,
+        spatial_metric, sqrt_det_spatial_metric));
+  }
+};
+
+}  // namespace M1Grey
+}  // namespace RadiationTransport
+
+namespace Actions {
+
+/// \ingroup ActionsGroup
+/// \brief Compute the source terms due to neutrino-matter interactions
+///
+/// \note `Metavariables` must specify a
+/// `list_of_neutrino_species`.
+///
+/// Uses:
+/// - DataBox: Items in
+/// RadiationTransport::M1Grey::ComputeM1HydroCoupling::argument_tags
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies: RadiationTransport::M1Grey::ComputeM1HydroCoupling::return_tags
+struct M1HydroCoupling {
```

Comment:
This is the same as the generic `MutateApply` action.

Reply:
Ok, will look for how to change it to MutateApply - I have never used that action before